cadastre/cadastre_common_base.py

```diff
@@ -218,7 +218,13 @@ def getConnectorFromUri(connectionParams: Dict[str, str]) -> 'DBConnector':
                 connectionParams['password']
             )
 
-        connector = PostGisDBConnector(uri)
+        # we need a fake DBPlugin object
+        # with connetonName and providerName methods
+        obj = QObject()
+        obj.connectionName = lambda: 'fake'
+        obj.providerName = lambda: 'postgres'
+
+        connector = PostGisDBConnector(uri, obj)
```

Comment:
```suggestion
        if Qgis.QGIS_VERSION_INT >= 31000:
            # we need a fake DBPlugin object
            # with connetonName and providerName methods
            obj = QObject()
            obj.connectionName = lambda: 'fake'
            obj.providerName = lambda: 'postgres'

            connector = PostGisDBConnector(uri, obj)
        else:
            connector = PostGisDBConnector(uri)
```

Si tu veux faire rétrocompatible, c'est suffisant, non?

Reply:
Oui c'est suffisant mais la version est >= 3.12
```
if Qgis.QGIS_VERSION_INT >= 31200:
```

Reply:
ok, j'ai corrigé